sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -172,7 +172,7 @@ case class Alias(child: Expression, name: String)(
           val builder = new MetadataBuilder().withMetadata(named.metadata)
           nonInheritableMetadataKeys.foreach(builder.remove)
           builder.build()
-
+        case structField: GetStructField => structField.metadata
```

Comment:
Don't we need to remove keys in `nonInheritableMetadataKeys`?

Reply:
Yes, you're right. I fixed it and added an additional test.